src/tracker/detector/detector.py

```diff
@@ -0,0 +1,22 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from src.config.config import DetectorConfig
+from src.image.image import ImageObject
+from src.tracker.detector.detection import Detection
+
+
+class Detector(ABC):
+    def __init__(self, cfg: DetectorConfig):
+        """
+        Initializes the detector with a DetectorConfig object that contains configuration for the model.
+        """
+        self.cfg = cfg
+        self.chosen_labels = cfg.chosen_labels
+
+    @abstractmethod
+    async def detect(self, image: ImageObject) -> List[Detection]:
+        """
+        Abstract method to be implemented by specific detector classes. Each detector must implement
+        this method to detect objects in the provided image.
+        """
```

Comment:
Okay I take back what I said/thought about deleting the Detection class you have here.  I guess it unifies the Viam stuff with whatever Torchvision is doing.  That being said, because they're unified, you should have a method here called filter() that takes in a List[Detections] and spits out a (probably smaller) List[Detections].

Reply:
It means you'll grab more when detecting at first but that's okay